pkg/aws/validations_internal_test.go

```diff
@@ -0,0 +1,238 @@
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/smithy-go"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+)
+
+var _ = Describe("Validations", func() {
+	Context("ValidatePeeringPrerequisites", testValidatePeeringPrerequisites)
+	Context("CheckVpcOverlap", testCheckVpcOverlap)
+	Context("DeterminePermissionError", testDeterminePermissionError)
+})
+
+func testDeterminePermissionError() {
+	Describe("Validate error input", func() {
+		When("error is nil", func() {
+			It("should return nil", func() {
+				err := determinePermissionError(nil, "")
+				Expect(err).To(BeNil())
+			})
+		})
+		When("called with an AWS DryRunOperation error", func() {
+			It("should return nil", func() {
+				err := smithy.GenericAPIError{
+					Code:    "DryRunOperation",
+					Message: "DryRunOperation",
+					Fault:   1,
+				}
+				operation := "check"
+				result := determinePermissionError(&err, operation)
+				Expect(result).To(BeNil())
+			})
+		})
+		When("called with an AWS UnauthorizedOperation error", func() {
+			It("should return an appropriate error", func() {
+				err := smithy.GenericAPIError{
+					Code:    "UnauthorizedOperation",
+					Message: "UnauthorizedOperation",
+					Fault:   1,
+				}
+				operation := "check"
+				result := determinePermissionError(&err, operation)
+				Expect(result).To(
+					MatchError(
+						MatchRegexp("no permission to " + operation),
+					),
+				)
+			})
+		})
+		When("called with a general error", func() {
+			It("should return an appropriate error", func() {
+				err := smithy.GenericAPIError{
+					Code:    "Generic Error",
+					Message: "Generic Error",
+					Fault:   1,
+				}
+				operation := ""
+				result := determinePermissionError(&err, operation)
+				Expect(result).To(
+					MatchError(
+						MatchRegexp("error while checking permissions for " + operation),
+					),
+				)
+			})
+		})
+	})
+}
+
+func testValidatePeeringPrerequisites() {
+	cloudA := newCloudTestDriver(infraID, region)
+	cloudB := newCloudTestDriver(targetInfraID, targetRegion)
+
+	Describe("Validate Validate Peering Prerequisites", func() {
+		When("trying to retrieve the VPC", func() {
+			It("should not retrieve the source VPC", func() {
```

Comment:
I assume this is testing no VPC found?

```suggestion
		When("no VPC exists for the source Cloud", func() {
			It("should return an error", func() {
```